src/devices/sound/flt_biquad.h

```diff
@@ -0,0 +1,80 @@
+// license:BSD-3-Clause
+// copyright-holders:K.Wilkins,Couriersud,Derrick Renaud,Frank Palazzolo,Jonathan Gevaryahu
+#ifndef MAME_SOUND_FLT_BIQUAD_H
+#define MAME_SOUND_FLT_BIQUAD_H
+
+#pragma once
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> filter_biquad_device
+
+class filter_biquad_device : public device_t, public device_sound_interface
+{
+public:
+	enum
+	{
+		LOWPASS1P = 0,
+		HIGHPASS1P,
+		LOWPASS,
+		HIGHPASS,
+		BANDPASS,
+		NOTCH,
+		PEAK,
+		LOWSHELF,
+		HIGHSHELF
+	};
+
+	filter_biquad_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock = 0);
+
+	// set up the filter with the specified filter parameters and return a pointer to the new device
+	filter_biquad_device& setup(int type, double fc, double q, double gain);
+	// update an existing instance with new filter parameters
+	void update_params(int type, double fc, double q, double gain);
+
+	// helper setup functions to create common filters representable by biquad filters
+	// Sallen-Key low-pass
+	filter_biquad_device& opamp_sk_lowpass_setup(double r1, double r2, double r3, double r4, double c1, double c2);
+	// TODO when needed: Sallen-Key band-pass
+	// TODO when needed: Sallen-Key high-pass
+
+	// Multiple-Feedback low-pass
+	filter_biquad_device& opamp_mfb_lowpass_setup(double r1, double r2, double r3, double c1, double c2);
+
+	// Multiple-Feedback band-pass
+	filter_biquad_device& opamp_mfb_bandpass_setup(double r1, double r2, double r3, double c1, double c2);
+
+	// Multiple-Feedback high-pass
+	filter_biquad_device& opamp_mfb_highpass_setup(double r1, double r2, double c1, double c2, double c3);
+
+
+protected:
+	// device-level overrides
+	virtual void device_start() override;
+
+	// sound stream update overrides
+	virtual void sound_stream_update(sound_stream &stream, std::vector<read_stream_view> const &inputs, std::vector<write_stream_view> &outputs) override;
+
+private:
+	void recalc();
+	void step();
+
+	sound_stream*  m_stream;
+	int            m_type;
```

Comment:
If you make the `enum` into a proper type, you can use it for this.

Reply:
This is now implemented.